core/commands/swarm.go

```diff
@@ -389,88 +366,38 @@ To see all resources that are close to hitting their respective limit, one can d
-- all           -- all limits actually being applied.
-- system        -- limits for the system aggregate resource usage.
-- transient     -- limits for the transient resource usage.
-- svc:<service> -- limits for the resource usage of a specific service.
-- proto:<proto> -- limits for the resource usage of a specific protocol.
-- peer:<peer>   -- limits for the resource usage of a specific peer.
-
+		Tagline: "Get actual limits reported by libp2p Resource Manager.",
+		LongDescription: `Get actual limits reported by libp2p Resource Manager.
 The output of this command is JSON.
-
-It is possible to use this command to inspect and tweak limits at runtime:
-
-	$ ipfs swarm limit system > limit.json
-	$ vi limit.json
-	$ ipfs swarm limit system limit.json
-
-Changes made via command line are persisted in the Swarm.ResourceMgr.Limits field of the $IPFS_PATH/config file.
 `},
-	Arguments: []cmds.Argument{
-		cmds.StringArg("scope", true, false, "scope of the limit"),
-		cmds.FileArg("limit.json", false, false, "limits to be set").EnableStdin(),
-	},
-	Options: []cmds.Option{
-		cmds.BoolOption(swarmResetLimitsOptionName, "reset limit to default"),
-	},
 	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
 		node, err := cmdenv.GetNode(env)
 		if err != nil {
 			return err
 		}
 
-		if node.ResourceManager == nil {
+		if _, ok := node.ResourceManager.(*network.NullResourceManager); ok {
 			return libp2p.ErrNoResourceMgr
 		}
 
-		scope := req.Arguments[0]
-
-		//  set scope limit to new values (when limit.json is passed as a second arg)
-		if req.Files != nil {
-			var newLimit rcmgr.ResourceLimits
-			it := req.Files.Entries()
-			if it.Next() {
-				file := files.FileFromEntry(it)
-				if file == nil {
-					return errors.New("expected a JSON file")
-				}
-
-				r := io.LimitReader(file, 32*1024*1024) // 32MiB
-
-				if err := json.NewDecoder(r).Decode(&newLimit); err != nil {
-					return fmt.Errorf("decoding JSON as ResourceMgrScopeConfig: %w", err)
-				}
-				return libp2p.NetSetLimit(node.ResourceManager, node.Repo, scope, newLimit)
-			}
-			if err := it.Err(); err != nil {
-				return fmt.Errorf("error opening limit JSON file: %w", err)
-			}
+		cfg, err := node.Repo.Config()
+		if err != nil {
+			return err
 		}
 
-		var result interface{}
-		switch _, reset := req.Options[swarmResetLimitsOptionName]; {
-		case reset:
-			result, err = libp2p.NetResetLimit(node.ResourceManager, node.Repo, scope)
-		case scope == "all":
-			result, err = libp2p.NetLimitAll(node.ResourceManager)
-		default:
-			// get scope limit
-			result, err = libp2p.NetLimit(node.ResourceManager, scope)
-		}
+		userRessourceOverrides, err := node.Repo.UserRessourceOverrides()
 		if err != nil {
 			return err
 		}
 
-		if base, ok := result.(rcmgr.BaseLimit); ok {
-			result = base.ToResourceLimits()
+		result, _, err := libp2p.LimitConfig(cfg.Swarm, userRessourceOverrides)
+		if err != nil {
+			return err
```

Comment:
I don't really know, this is the result of merging two's Antonio PRs, I think there might be some weird details where you can't query that easily from a running ressource manager idk. I'll look into it.

Reply:
There is no good API to get the required limit object, you can view individual peers and scopes but can't enumerate them. This is miles more readable and more consises than reading it by hand from the object.

Reply:
Ok, so to be clear, are you saying that there isn't an API to query go-libp2p to find out "what are the limits that you are enforcing"? That seems like a gap and seems odd that we're doing hacks around that gap vs. getting go-libp2p to expose that info.

If we really can't get that from libp2p I think:
1. We need to comment this
2. Rather than recreating what we passed to libp2p, can we just store the object in memory and then retrieve it? That will be more accurate since it's (small chance) that the limits.json file has been updated.

That said, stepping back, this is concerning the swarm stats command which I think we can get rid of (it's a subset of the swarm resources command). Although I guess we have the same problem there as well.

Reply:
We discussed this verbally 2023-03-02.  
I am going to create a issue against go-libp2p for a better API and @Jorropo can add a FIXME with a link to that issue.

Reply:
go-libp2p issue: https://github.com/libp2p/go-libp2p/issues/2166

Reply:
This code was removed as part of: https://github.com/ipfs/kubo/pull/9680#discussion_r1123616841